pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -85,6 +86,19 @@ func (pl *PodTopologySpread) Name() string {
 	return Name
 }
 
+// Pod topology spread is not localized to a pod and node, so we cannot
+// sign pods that have topology spread constraints, either explicit or
+// defaulted.
+func (pl *PodTopologySpread) SignPod(ctx context.Context, pod *v1.Pod) ([]fwk.SignFragment, *fwk.Status) {
+	if len(pl.defaultConstraints) > 0 || len(pod.Spec.TopologySpreadConstraints) > 0 {
```

Comment:
Should we only check `len(pl.defaultConstraints) > 0` when a pod is from the Services, Replication Controllers, Replica Sets, or Stateful Sets? Otherwise, this would be another thing that you have to change when you want to get the optimization from this feature since defaultConstraints is non-empty by default.